### PR TITLE
Configure GitHub Actions to deploy Router Worker

### PR DESCRIPTION
**Pull request by GregBrimble**
Will conflict at `packages/workers-shared/package.json#scripts.deploy` with https://github.com/cloudflare/workers-sdk/pull/6542

## What this PR solves / how to test

Fixes #[insert GH or internal issue number(s)].

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: GH actions change
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: GH actions change
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: internal facing change
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: internal facing change

### CODE REVIEW

**Review by CarmenPopoviciu — APPROVED** [review]

**Review by edmundhung — APPROVED** [review]

File: packages/workers-shared/package.json [hunk 1]
```diff
@@ -31,6 +31,8 @@
 		"check:type": "pnpm run check:type:tests && tsc",
 		"check:type:tests": "tsc -p ./asset-worker/tests/tsconfig.json",
 		"clean": "rimraf dist",
+		"deploy": "pnpm run deploy:router-worker && pnpm run deploy:asset-worker",
```
Comment (petebacondarwin) [inline packages/workers-shared/package.json:34]: Am I missing something but I don't see the `deploy:asset-worker` script definition here...

File: packages/workers-shared/package.json [hunk 2]
```diff
@@ -31,6 +31,8 @@
...
+		"deploy:router-worker": "CLOUDFLARE_API_TOKEN=$WORKERS_DEPLOY_AND_CONFIG_CLOUDFLARE_API_TOKEN wrangler versions upload --experimental-versions -c router-worker/wrangler.toml",
```
Comment (petebacondarwin) [inline packages/workers-shared/package.json:35]: I thought @GregBrimble said that we needed to provide some additional metadata for this deployment?

File: packages/workers-shared/package.json (hunk 1 above)
Comment (CarmenPopoviciu) [reply packages/workers-shared/package.json:34]: yeah, we need to merge https://github.com/cloudflare/workers-sdk/pull/6542 first. I will handle that

Comment (CarmenPopoviciu) [reply packages/workers-shared/package.json:34]: fixed

File: packages/workers-shared/package.json (hunk 2 above)
Comment (CarmenPopoviciu) [reply packages/workers-shared/package.json:36]: correct. added in https://github.com/cloudflare/workers-sdk/pull/6543/commits/56a3866b7291a0f932718f7b9b55186c4d10e8be

**Review by petebacondarwin — APPROVED** [review]
LGTM!

**Pull request merged**